_posts/en/newsletters/2022-12-07-newsletter.md

```diff
@@ -0,0 +1,238 @@
+  a0link="https://bitcoincore.reviews/26152#30"
+
+  q1=”What does a transaction’s \"cluster\" consist of?”
+  a1=”A transaction's cluster is the set consisting of itself and all
+      \"connected\" transactions. A sibling, i.e. a child of
+      your parent, who is neither your ancestor nor your descendant,
+      would be part of your cluster.
+  a1link="https://bitcoincore.reviews/26152#72"
+
+  q2="This PR introduces `MiniMiner` which duplicates some of the
+      actual miner's algorithms; would it have been better to
+      unify these two implementations through refactoring?"
+  a2="We only need to operate on a cluster and not the entire mempool,
+      and don't need to apply any of the checks that `BlockAssembler`
+      does. It was also suggested to do this calculation without
+      holding the mempool lock. We'd also need to change the block
+      assembler to be tracking bump fees rather than building the
+      block template; the amount of refactoring necessary was
+      equivalent to rewriting."
+  a2link="https://bitcoincore.reviews/26152#94"
+
+  q3="Why does the MiniMiner require an entire cluster? Why can’t it
+      just use the union of each transaction’s ancestor sets?"
+  a3="Some of the ancestors may already have been paid for by some of
+      their other descendants; we don't have to pay with our transaction.
+      So we need to include these other descendants in our calculations."
+  a3link="https://bitcoincore.reviews/26152#129"
+
+  q4="If transaction X has a higher _ancestor feerate_ than independent
+      transaction Y, is it possible for a miner to prioritize Y
+      over X (that is, mine Y before X)?"
+  a4="Yes. If some of Y's low-feerate ancestors have _other_ decendants that
+      are high feerate, Y doesn't need to \"pay\" for those ancestors;
+      we can, in effect, \"remove\" those ancestors from Y's ancestor set,
+      which would then increase Y's _effective_ ancestor feerate."
+  a4link="https://bitcoincore.reviews/26152#169"
+
+  q5="Can `CalculateBumpFees()` overestimate, underestimate, both, or
+      neither? By how much?"
+  a5="It will overestimate if two outputs with overlapping ancestry are
+      chosen, since they're bumped independently (without taking the shared
+      ancestry into account). Another way is if an ancestor acquires a
+      high feerate child after we've calculated the bump fees. The
+      participants couldn't think of a way bump fees could be
+      underestimated, but the question was left open."
+  a5link="https://bitcoincore.reviews/26152#194"
+
+  q6="The `MiniMiner` is given a list of UTXOs (outpoints) that the wallet
+      might be interested in spending. Given an outpoint, what are its 4
+      possible states?"
+  a6="It could be a confirmed, or unconfirmed (in the mempool), or unconfirmed
+      but already being spent by an existing transaction in the mempool,
+      or it could be an outpoint that we've never heard of."
```

Comment:
I think the chat actually concluded with 5 states https://bitcoincore.reviews/26152-2#l-54 (even though the question asked for 4)
```suggestion
  a6="It could be a confirmed and unspent, confirmed but already spent
      by an existing transaction in the mempool, unconfirmed (in the
      mempool) and unspent, or unconfirmed
      but already spent by an existing transaction in the mempool,
      or it could be an outpoint that we've never heard of."
```